### PR TITLE
[WIP] Integrate login

### LINKED ISSUE #198: the webapp must be protected by a login page.
An user must be authenticated and authorized to see the resource he's trying to see.

I started to work on this here: https://github.com/pierrealixt/ford3/tree/login-198

Screen Shot 2019-05-16 at 08 49 52

### PR DESCRIPTION
**Pull request by boney-bun**
fix #198

### CODE REVIEW

File: django_project/core/base_templates/index.html [hunk 1]
```diff
@@ -34,7 +34,24 @@
         <nav class="navbar">
             <div class='container'>
                 <p class="open-edu-logo"><span class='text-white'>Open</span><span class='text-blue'>Edu</span></p>
-                <a class='text-white' href='/site-admin'>Administration</a>
+            <div class="col-md-8">
+            {% if user.is_authenticated %}
+                <p class="text-white text-right">
+                    Welcome, <b>{{ user.username }}</b>
```
Comment (pierrealixt) [inline django_project/core/base_templates/index.html]: remove the `welcome`

File: django_project/ford3/templates/base.html [hunk 2]
```diff
@@ -5,7 +5,24 @@
 {% block navbar %}
 <nav class="navbar pl-0">
         <div class="open-edu-logo col-md-2 text-center px-0"><span class='text-white'>Open</span><span class='text-blue'>Edu</span></div>
-        <a class='text-white text-right mr-0' href='/site-admin'>Administration</a>
+        <div class="col-md-8">
+        {% if user.is_authenticated %}
+            <p class="text-white text-right">
+                Welcome, <b>{{ user.username }}</b>
```
Comment (pierrealixt) [inline django_project/ford3/templates/base.html]: remove the `welcome`

File: django_project/ford3/templates/registration/login.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                </div>
+            </div>
+            <div class="row">
+                <h4 class="mx-auto text-white">Login</h4>
+            </div>
+              <form method="post">
+                {% csrf_token %}
+                      {% for field in form  %}
+                          <div class="row my-3">
+                              <div class="w-100 mx-3">{{ field }}</div>
```
Comment (pierrealixt) [inline django_project/ford3/templates/registration/login.html]: placeholder?
label?
Screen Shot 2019-05-17 at 09 05 45

**Comment by boney-bun** [comment]
PIA:
ford3#417

<UPDATED>
4 questions in mind :
1. i think we need a burger menu on the top nav bar. next question is what are the list of options?
2. after a user logged out, which page to redirect to? 
3. i still need to add *forgot password* and *register new user* in the login page. we need to determine the use case for the two, such as: do the user automatically added to the system or an admin is required to approve the user? 
4. i tried to change the color of user icon by using online photo editor, but it can't import the image into svg. the end result is not good. can someone change the color and keep it into svg format?

**Comment by pierrealixt** [comment]
@boney-bun 
> i think we need a burger menu on the top nav bar. next question is what are the list of options?

A burger menu to show what?
For now your PR does what the issue is asking: adding a login view.

> after a user logged out, which page to redirect to?

Back to the login view I guess.

> i still need to add forgot password and register new user in the login page. we need to determine the use case for the two, such as: do the user automatically added to the system or an admin is required to approve the user?

This is not the scope of this PR.

File: django_project/core/base_templates/index.html (hunk 1 above)
Comment (boney-bun) [reply django_project/core/base_templates/index.html]: yes, it's a sample only. do we have mockup for this part?
i imagine we will have a burger kind of menu contains, at the very least, logout option. what else do we need here? dashboard? link to admin page?

**Comment by pierrealixt** [comment]
Screen Shot 2019-05-17 at 09 07 09
Login link in the navbar is not pretty.

**Comment by pierrealixt** [comment]
@boney-bun 

> 4. i tried to change the color of user icon by using online photo editor, but it can't import the image into svg. the end result is not good. can someone change the color and keep it into svg format?

Can't you find an white user icon?
see https://www.flaticon.com/

**Comment by pierrealixt** [comment]
There is an issue with the navbar between the landing page and the rest of the app.
See top right
bugnavbar

File: django_project/core/base_templates/index.html (hunk 1 above)
Comment (pierrealixt) [reply django_project/core/base_templates/index.html]: No. but for now, the username + a logout link is ok.

**Review by pierrealixt — COMMENTED** [review]
Please see my comments

**Comment by boney-bun** [comment]
0a4d00d: 
ford3#417_rev

@pierrealixt do we still need labels if we have placeholders already?
Screen Shot 2019-05-20 at 4 35 38 PM

File: django_project/ford3/forms/custom_auth_form.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from django import forms
+from django.contrib.auth.forms import AuthenticationForm
+from django.forms.widgets import PasswordInput, TextInput
+
+
+class CustomAuthForm(AuthenticationForm):
+    username = forms.CharField(widget=TextInput(attrs={'placeholder': 'Institution'}))
```
Comment (pierrealixt) [inline django_project/ford3/forms/custom_auth_form.py]: Remove `institution`, we'll ask for the user email.

File: django_project/ford3/templates/registration/login.html [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                          </div>
+                      {% endfor  %}
+                <div class="row">
+                    <div class="w-100 text-right mx-3">
+                        <a href="#">Forgot password?</a>
```
Comment (pierrealixt) [inline django_project/ford3/templates/registration/login.html]: remove from PR if not implemented.
we'll do it later.

File: django_project/ford3/templates/registration/login.html [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+                <div class="row">
+                    <button class="edu-button edu-button-orange mx-3" type="submit">Login</button>
+                </div>
+                <div class="row my-3">
+                    <div class="w-100 text-center">
+                        <a href="#">NOT YET REGISTERED? SIGN UP</a>
+                    </div>
+                </div>
+              </form>
+        </div>
```
Comment (pierrealixt) [inline django_project/ford3/templates/registration/login.html:41]: remove from PR if not implemented.
we'll do it later.

**Comment by boney-bun** [comment]
the new login:
Screen Shot 2019-05-20 at 5 12 23 PM

**Comment by boney-bun** [comment]
@pierrealixt the new login:
Screen Shot 2019-05-21 at 12 53 02 PM

what do you think?

**Review by pierrealixt — COMMENTED** [review]
LGTM

**Review by pierrealixt — APPROVED** [review]
LGTM

**Pull request merged**